api/src/main/java/org/apache/iceberg/expressions/StrictMetricsEvaluator.java

```diff
@@ -144,6 +146,35 @@ public Boolean or(Boolean leftResult, Boolean rightResult) {
       return ROWS_MIGHT_NOT_MATCH;
     }
 
+    @Override
+    @SuppressWarnings("checkstyle:CyclomaticComplexity")
+    public <T> Boolean isNaN(BoundReference<T> ref) {
+      int id = ref.fieldId();
+
+      if (nanCounts != null && nanCounts.containsKey(id) &&
```

Comment:
can refactor with `containsNaNOnly` as discussed above, and the checkstyle issue should not show up anymore.